hbase-server/src/main/java/org/apache/hadoop/hbase/master/janitor/CatalogJanitor.java

```diff
@@ -252,14 +265,11 @@ private boolean cleanMergeRegion(final RegionInfo mergedRegion, List<RegionInfo>
           .stream().map(r -> RegionInfo.getShortNameToLog(r)).collect(Collectors.joining(", ")),
         mergedRegion);
       ProcedureExecutor<MasterProcedureEnv> pe = this.services.getMasterProcedureExecutor();
-      pe.submitProcedure(
-        new GCMultipleMergedRegionsProcedure(pe.getEnvironment(), mergedRegion, parents));
-      for (RegionInfo ri : parents) {
-        // The above scheduled GCMultipleMergedRegionsProcedure does the below.
```

Comment:
GCMultipleMergedRegionsProcedure does this. It is not needed here too. I think it better to let the procedure do it. We expect the procedure to take care of such details, so let's in fact do that.
Keeping this code around would be harmless, though.